test/view_test.go

```diff
@@ -0,0 +1,473 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	driver "github.com/arangodb/go-driver"
+)
+
+// ensureArangoSearchView is a helper to check if an arangosearch view exists and create it if needed.
+// It will fail the test when an error occurs.
+func ensureArangoSearchView(ctx context.Context, db driver.Database, name string, options *driver.ArangoSearchViewProperties, t testEnv) driver.ArangoSearchView {
+	v, err := db.View(ctx, name)
+	if driver.IsNotFound(err) {
+		v, err = db.CreateArangoSearchView(ctx, name, options)
+		if err != nil {
+			t.Fatalf("Failed to create arangosearch view '%s': %s", name, describe(err))
+		}
+	} else if err != nil {
+		t.Fatalf("Failed to open view '%s': %s", name, describe(err))
+	}
+	result, err := v.ArangoSearchView()
+	if err != nil {
+		t.Fatalf("Failed to open view '%s' as arangosearch view: %s", name, describe(err))
+	}
+	return result
+}
+
+// tryCreateArangoSearchLink is a helper that adds a link to a view and collection.
```

Comment:
This comment is confusing.
The method says "create" but the comment says "add". Which is correct?